### DISCUSSION TITLE
How to use context.waitUntil in route handler?

### DISCUSSION THREAD

**Discussion by s1973** [vercel/next.js#50441] [Help]
### Summary

When using pages router, we can use context on api routers:
```ts
import { NextResponse } from 'next/server';
import type { NextFetchEvent, NextRequest } from 'next/server';
 
export const config = {
  runtime: 'edge',
};
 
const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
 
async function getAlbum() {
  const res = await fetch('https://jsonplaceholder.typicode.com/albums/1');
  await wait(10000);
  return res.json();
}
 
export default function MyEdgeFunction(
  request: NextRequest,
  context: NextFetchEvent,
) {
  context.waitUntil(getAlbum().then((json) => console.log({ json })));
 
  return NextResponse.json({
    name: `Hello, from ${request.url} I'm an Edge Function!`,
  });
}
```
while when using app router, we got this from doc:
> Currently, the only value of context is params, which is an object containing the [dynamic route parameters](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) for the current route.

so I am wondering how to use NextFetchEvent with route handler? 

Thanks for any response in advance!

**Comment by ShantanuNair** [comment]
@s1973 Looks like Vercel docs are still lacking here. Did you figure this out?

**Reply by s1973** [reply]
Haven't solved yet either, I have no clue unless turn back to Pages Router

**Comment by oagix** [comment]
I'd love to know this too.

**Comment by matannahmani** [comment]
Any updates?

**Comment by s1973** [comment]
No result yet but might worth tracking here https://github.com/vercel/next.js/issues/50522

**Comment by weleo** [comment]
[waitUntil()](https://vercel.com/docs/functions/edge-functions/edge-functions-api#waituntil) From the docs, it is currently not working in the app router. As mentioned before, keeping an eye on:

> No result yet but might worth tracking here #50522

**Comment by Thinkscape** [comment]
Good news:

https://vercel.com/changelog/waituntil-is-now-available-for-vercel-functions
https://vercel.com/docs/functions/functions-api-reference#waituntil
